### PR TITLE
[12.0][MIG] project_task_dependency

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Standard migration of project_task_dependency to v12

### CODE REVIEW

**Review by tarteo — APPROVED** [review]

File: project_task_dependency/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+* Dennis Sluijk <d.sluijk@onestein.nl>
```
Comment (pedrobaeza) [inline project_task_dependency/readme/CONTRIBUTORS.rst]: You can add yourself here, and I also did some optimization changes in v11, so you can add me as well.

**Review by pedrobaeza — APPROVED** [review]

File: project_task_dependency/readme/CONTRIBUTORS.rst (hunk 1 above)
Comment (AdriaGForgeFlow) [reply project_task_dependency/readme/CONTRIBUTORS.rst]: Done :+1:

File: project_task_dependency/tests/test_project_task_dependency.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        with self.assertRaises(ValidationError):
+            self.task1.write({
+                'dependency_task_ids': [(6, 0, [self.task3.id])]
+            })
+
+    def test_copy(self):
+        new_project = self.project1.copy({
+            'name': 'Nice Project Test Dependencies One Second'
+        })
+        task1 = new_project.tasks.filtered(lambda t: t.name == '2')
```
Comment (tafaRU) [inline project_task_dependency/tests/test_project_task_dependency.py]: Wouldn't be better to use `task2`?

**Review by tafaRU — CHANGES_REQUESTED** [review]

File: project_task_dependency/tests/test_project_task_dependency.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        task3 = new_project.tasks.filtered(lambda t: t.name == '3')
+        self.assertEqual(task3.dependency_task_ids[0].name, '2')
+        new_project = self.project2.copy({
+            'name': 'Nice Project Test Dependencies Two Second'
+        })
+        task4 = new_project.tasks.filtered(lambda t: t.name == '4')
+        self.assertEqual(task4.dependency_task_ids[0].id, self.task2.id)
+
+    def test_regression_copy(self):
+        old_count = self.env['project.task.copy.map'].search(
```
Comment (Nikul-Chaudhary) [inline project_task_dependency/tests/test_project_task_dependency.py]: @ageficent IMO use `search_coun`t mehtod

File: project_task_dependency/tests/test_project_task_dependency.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            [('old_task_id', '=', self.task4.id)], count=True
+        )
+        self.task4.copy()
+        new_count = self.env['project.task.copy.map'].search(
```
Comment (Nikul-Chaudhary) [inline project_task_dependency/tests/test_project_task_dependency.py]: same here

**Review by Nikul-Chaudhary — CHANGES_REQUESTED** [review]
Improve Code

File: project_task_dependency/tests/test_project_task_dependency.py (hunk 2 above)
Comment (AdriaGForgeFlow) [reply project_task_dependency/tests/test_project_task_dependency.py]: Done :+1:

File: project_task_dependency/tests/test_project_task_dependency.py (hunk 3 above)
Comment (AdriaGForgeFlow) [reply project_task_dependency/tests/test_project_task_dependency.py]: Done :+1:

**Review by tafaRU — APPROVED** [review]
Just an advice: 

use `git commit --fixup <commit>` to automatically marks your commit as a fix of a previous commit

See https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html for further reference.

**Comment by pedrobaeza** [comment]
@tafaRU, FYI, there's no need of more "fixup" thanks to GitHub, that now includes the diff between different forced pushes, so it's better now to amend instead of fixing up, as you don't have to reupload the merged commits once the review has finished:

seleccion_029

**Comment by tafaRU** [comment]
@pedrobaeza, good, one can always learn from the best!
Thank you for the advice.

**Comment by AdriaGForgeFlow** [comment]
Thanks for the advice. I'll take it into account for the future :+1:

**Comment by AdriaGForgeFlow** [comment]
@Nikul-Chaudhary Can you update the review please?

**Review by Nikul-Chaudhary — APPROVED** [review]
Code Review LGTM :+1:

**Pull request merged**